monai/data/image_reader.py

```diff
@@ -0,0 +1,348 @@
+            is already in memory and no need to read from file again, default is None.
+
+    """
+
+    def __init__(self, img: Optional[Any] = None) -> None:
+        self.img = img
+        self._suffixes: Optional[Sequence[str]] = None
+
+    def get_suffixes(self) -> Sequence[str]:
+        """
+        Get the supported image file suffixes of current reader.
+        Default is None, support all kinds of image format.
+
+        """
+        return self._suffixes
+
+    def verify_suffix(self, filename: str) -> bool:
+        """
+        Verify whether the specified file or files match supported suffixes.
+        If supported suffixes is None, skip the verification and return True.
+
+        Args:
+            filename: file name or a list of file names to read.
+                if a list of files, verify all the subffixes.
+        """
+
+        if self._suffixes is None:
+            return True
+
+        supported_format: bool = True
+        filenames: Sequence[str] = ensure_tuple(filename)
+        for name in filenames:
+            suffixes: Sequence[str] = name.split(".")
+            passed: bool = False
+            for i in range(len(suffixes) - 1):
+                if ".".join(suffixes[-(i + 2) : -1]) in self._suffixes:
+                    passed = True
+                    break
+            if not passed:
+                supported_format = False
+                break
+
+        return supported_format
+
+    def uncache(self) -> None:
+        """
+        Release image object and other cache data.
+
+        """
+        self.img = None
+
+    @abstractmethod
+    def read(self, filename: str) -> Union[Sequence[Any], Any]:
+        """
+        Read image data from specified file or files and save to `self.img`.
+        Note that it returns the raw data, so different readers return different image data type.
+
+        Args:
+            filename: file name or a list of file names to read.
+
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
+    @abstractmethod
+    def get_data(self) -> Tuple[np.ndarray, Dict]:
+        """
+        Extract data array and meta data from loaded image and return them.
+        This function must return 2 objects, first is numpy array of image data, second is dict of meta data.
+
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
+
+class ITKReader(ImageReader):
+    """
+    Load medical images based on ITK library.
+    All the supported image formats can be found:
+    https://github.com/InsightSoftwareConsortium/ITK/tree/master/Modules/IO
+
+    Args:
+        img: image to initialize the reader, this is for the usage that the image data
+            is already in memory and no need to read from file again, default is None.
+        keep_axes: default to `True`. if `False`, the numpy array will have C-order indexing.
```

Comment:
I wonder if a more intuitive argument name could be used instead of keep_axis.  Perhaps "c_order_axis_indexing" or "numpy_axis_indexing" or something descriptive?   I realize ITK calls it keep_axes, but maybe MONAI doesn't need to use that name?

Reply:
updated to `c_order_axis_indexing` in the latest commit.
Thanks.